server/src/main/java/org/opensearch/common/ResourceType.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Enum to specify the system level resources
+ */
+@ExperimentalApi
+public enum ResourceType {
+    CPU("cpu"),
+    HEAP_ALLOCATIONS("heap_allocations");
```

Comment:
Why not JVM or Memory?

Reply:
Because they seem ambiguous and doesn't convey the right intent. as we are tracking the heap_allocations not the memory as the memory itself is ambiguous.

Reply:
While HEAP_ALLOCATIONS might be the most precise, I would also like some consistency across QueryInsights and Sandboxing. So, maybe go with MEMORY